### PR TITLE
dev: clean up

### PR DESCRIPTION
**Pull request by ldez**
Review from #4457 by @Antonboom

### CODE REVIEW

File: scripts/website/expand_templates/main.go [hunk 1]
```diff
@@ -92,29 +93,35 @@ type latestRelease struct {
...
 
 func getLatestVersion() (string, error) {
-	req, err := http.NewRequest( //nolint:noctx
-		http.MethodGet,
-		"https://api.github.com/repos/golangci/golangci-lint/releases/latest",
-		http.NoBody,
-	)
+	endpoint := "https://api.github.com/repos/golangci/golangci-lint/releases/latest"
+
+	//nolint:noctx
```
Comment (alexandear) [inline scripts/website/expand_templates/main.go:98]: Could we change `getLatestVersion` to accept context and remove this `//nolint:noctx`?

```go
func getLatestVersion(ctx context.Context) (string, error) {
```

We can pass at least `context.Background()` when calling `getLatestVersion`.

File: scripts/website/expand_templates/main.go [hunk 2]
```diff
@@ -40,7 +41,7 @@ func rewriteDocs(replacements map[string]string) error {
 			return processDoc(path, replacements, madeReplacements)
 		})
 	if err != nil {
-		return fmt.Errorf("failed to walk dir: %w", err)
+		return fmt.Errorf("walk dir: %w", err)
```
Comment (alexandear) [inline scripts/website/expand_templates/main.go:44]: There are a lot of places where we can remove the string "failed to " from an error message: https://github.com/search?q=repo%3Agolangci%2Fgolangci-lint+%22fmt.Errorf%28%5C%22failed+to+%22&type=code
However, this should be done as a separate refactoring PR.

**Review by alexandear — APPROVED** [review]
LGTM

File: scripts/website/expand_templates/main.go (hunk 1 above)
Comment (ldez) [reply scripts/website/expand_templates/main.go:98]: it's useless to use `context.Background()` if there is nothing based on this context.

File: scripts/website/expand_templates/main.go (hunk 2 above)
Comment (ldez) [reply scripts/website/expand_templates/main.go:44]: It was an element of the review but I don't want to change all the error messages (please don't create a PR for that).

File: scripts/website/expand_templates/main.go (hunk 1 above)
Comment (alexandear) [reply scripts/website/expand_templates/main.go:98]: Yeah, you are right. I think we can add a timeout, e.g. 2 seconds. Sometimes even GitHub has [issues](https://www.githubstatus.com/history).

File: scripts/website/expand_templates/main.go [hunk 3]
```diff
@@ -92,29 +93,35 @@ type latestRelease struct {
...
+		return "", fmt.Errorf("read a body for the latest tag: %w", err)
 	}
+
 	release := latestRelease{}
+
 	err = json.Unmarshal(body, &release)
 	if err != nil {
-		return "", fmt.Errorf("failed to unmarshal the body for the latest tag: %w", err)
+		return "", fmt.Errorf("unmarshal the body for the latest tag: %w", err)
 	}
```
Comment (alexandear) [inline scripts/website/expand_templates/main.go:123]: We can refactor to `json.Decoder`, which will make it four lines shorter (3 for handling `io.ReadAll`, and 1 for importing `io`):
```go
	release := latestRelease{}
	err = json.NewDecoder(resp.Body).Decode(&release)
	if err != nil {
		return "", fmt.Errorf("failed to decode the body for the latest tag: %w", err)
	}
```

File: scripts/website/expand_templates/main.go (hunk 1 above)
Comment (ldez) [reply scripts/website/expand_templates/main.go:98]: the timeout for an `htttp.Client` should be set on the client, not on the request.

File: scripts/website/expand_templates/main.go (hunk 3 above)
Comment (ldez) [reply scripts/website/expand_templates/main.go:123]: One of the differences between the `io.ReadAll` + `json.Unmarshal` and `json.NewDecoder().Decode` is the quality of the error.
The errors of `json.NewDecoder().Decode`, if the problem is related to the read of the body, are less relevant than `io.ReadAll`.

**Comment by Antonboom** [comment]
❤️

**Pull request merged**